In [1]:
#Importing libraries

import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix
from sklearn.preprocessing import StandardScaler
from sklearn.ensemble import RandomForestClassifier
from sklearn.feature_selection import SelectFromModel

In [2]:
# Importing data

LendingData = pd.read_csv('Resources/lending_data.csv')
LendingData.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0

In [3]:
# Splitting data into X_train, X_test, Y_train, Y_test

X = LendingData[['loan_size','interest_rate','borrower_income', 'debt_to_income', 'num_of_accounts', 'derogatory_marks', 'total_debt']]
Y = LendingData['loan_status']
print("X Values: ", X.shape)
print("Y Values: ", Y.shape)

X Values:  (77536, 7)
Y Values:  (77536,)


In [4]:
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, random_state=1)

In [5]:
# Train a Logistic Regression model print the model score

classifier = LogisticRegression()
classifier.fit(X_train, Y_train)
classifier

LogisticRegression()

In [6]:
print(f"Training Score as per Logistic Regression model: {classifier.score(X_train, Y_train)}")
print(f"Testing Score as per Logistic Regression model : {classifier.score(X_test, Y_test)}")

Training Score as per Logistic Regression model: 0.9921240885954051
Testing Score as per Logistic Regression model : 0.9918489475856377


In [7]:
# Train a Random Forest Classifier model and print the model score

scaler = StandardScaler().fit(X_train)
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [8]:
clf = RandomForestClassifier(random_state=1, n_estimators=500).fit(X_train_scaled, Y_train)
print(f'Training Score with a Random Forest Classifier model: {clf.score(X_train_scaled, Y_train)}')
print(f'Testing Score with a Random Forest Classifier model : {clf.score(X_test_scaled, Y_test)}')

Training Score with a Random Forest Classifier model: 0.9975409272252029
Testing Score with a Random Forest Classifier model : 0.9917457697069748


### Conclusion
##### There is not a big difference in the performance between both models for the lending data provided. However, Random Forest Classifier had the higher Train Score of 0.9975 while Logistic Regression Model had a score of 0.9921.

##### Random Forest Classifier performed better than Logistic Regression classification because a random selection of features is selected each time which provides a greater ensemble to aggregate over producing a more accurate predicton